pkg/config/config.go

```diff
@@ -69,3 +80,84 @@ func (config *Config) SaveToFile(filename string) error {
 
 	return nil
 }
+
+// TargetMatch represents a pattern match
+type TargetMatch struct {
+	// Garden is the matched Garden
+	Garden string
+	// Project is the matched Project
+	Project string
+	// Namespace is the matched Namespace, can be used to find the related project
+	Namespace string
+	// Garden is the matched Shoot
+	Shoot string
+}
+
+func contains(values []string, value string) bool {
+	for _, v := range values {
+		if v == value {
+			return true
+		}
+	}
+
+	return false
+}
+
+// FindGarden returns name of Garden with provided name or first that includes the provided name in the list of aliases
```

Comment:
I didn't really understand this doc string tbh. It sounds more complicated than it is